### ISSUE TITLE
Migrating to secondary database fails

### ISSUE THREAD

**Issue by PatTheMav** [modoboa/modoboa#1682]
# Impacted versions

* Modoboa: 1.13.1
* installer used: Yes
* Webserver: Nginx

# Steps to reproduce
1. Have Modoboa instance run on PostgreSQL
1. Add new MySQL database entry with identical credentials (database is named `mysql`)
1. Execute `python manage.py migrate --database=mysql`

# Current behavior

Django starts the migration process by applying "all migrations":

```
Operations to perform:
  Apply all migrations: admin, auth, authtoken, contenttypes, core, dnstools, lib, limits, modoboa_amavis, modoboa_contacts, modoboa_postfix_autoreply, modoboa_radicale, relaydomains, reversion, sessions, sites, transport
```

The process starts successfully:

```
Running migrations:
  Applying transport.0001_initial... OK
  Applying contenttypes.0001_initial... OK
  Applying contenttypes.0002_remove_content_type_name... OK
  Applying auth.0001_initial... OK
  Applying core.0001_initial... OK
  Applying admin.0001_initial... OK
  Applying admin.0002_migrate_from_modoboa_admin... OK
  Applying relaydomains.0001_initial... OK
  Applying relaydomains.0002_migrate_from_modoboa_admin_relaydomains... OK
  Applying relaydomains.0003_auto_20170116_1508... OK
```

During `Applying relaydomains.0004_auto_20161105_1424`, a `django.db.utils.ProgrammingError` occurs (roughly translated from German):

```
ERROR:  Relation »postfix_relay_domains_relaydomain« doesn't exist
LINE 1: ...lay_domains_relaydomain"."last_modification" FROM "postfix_r...
```

# Expected behavior

Expected behavior would be for Django to fully migrate the current Postgres state into the new MySQL (as far as possible, it's acceptable if it breaks because of driver-specific constraints).

*Note*: I'm not a Django expert, but afaik the `relaydomains` table has been renamed `transports` in a later Modoboa update. During `migrate` it seems to successfully create the table in the MySQL table, but then fails to find it in PostgreSQL table (as is it's not there anymore).

Also, the Modoboa installation itself runs without issues and after manually copying over all tables and recreating accounts in permissions in MySQL, Modoboa runs without issues on the new database as well.

# Video/Screenshot link (optional)

Here's the output of `python manage.py showmigrations`:

```
admin
 [X] 0001_initial
 [X] 0002_migrate_from_modoboa_admin
 [X] 0003_auto_20151118_1215
 [X] 0004_auto_20160914_0907
 [X] 0005_auto_20161026_1003
 [X] 0006_auto_20161104_1900
 [X] 0007_auto_20161104_1915
 [X] 0008_domain_enable_dns_checks
 [X] 0009_auto_20170215_0948
 [X] 0010_auto_20170215_1328
 [X] 0011_domain_transport
 [X] 0012_auto_20180112_1658
 [X] 0013_auto_20180124_2311
 [X] 0014_auto_20181017_1628
auth
 [X] 0001_initial
 [X] 0002_alter_permission_name_max_length
 [X] 0003_alter_user_email_max_length
 [X] 0004_alter_user_username_opts
 [X] 0005_alter_user_last_login_null
 [X] 0006_require_contenttypes_0002
 [X] 0007_alter_validators_add_error_messages
 [X] 0008_alter_user_username_max_length
authtoken
 [X] 0001_initial
 [X] 0002_auto_20160226_1747
contenttypes
 [X] 0001_initial
 [X] 0002_remove_content_type_name
core
 [X] 0001_initial
 [X] 0002_delete_extension
 [X] 0003_user_master_user
 [X] 0004_auto_20150728_1228
 [X] 0005_user_language
 [X] 0006_auto_20151114_1518
 [X] 0007_auto_20151116_1349
 [X] 0008_localconfig
 [X] 0009_auto_20161026_1003
 [X] 0010_auto_20161026_1011
 [X] 0011_auto_20161104_2217
 [X] 0012_auto_20170215_1034
 [X] 0013_auto_20170707_1058
 [X] 0014_auto_20171010_1746
 [X] 0015_auto_20180112_1642
 [X] 0016_auto_20180713_1014
dnstools
 [X] 0001_initial
lib
 [X] 0001_initial
 [X] 0002_rename_parameters
 [X] 0003_rename_parameters
 [X] 0004_auto_20151114_1409
 [X] 0005_auto_20160416_1449
 [X] 0006_auto_20161104_2206
limits
 [X] 0001_initial
 [X] 0002_auto_20151114_1518
 [X] 0003_auto_20160413_1046
 [X] 0004_auto_20160413_1312
 [X] 0005_auto_20160415_1654
 [X] 0006_auto_20170216_1112
modoboa_amavis
 [X] 0001_initial
modoboa_contacts
 [X] 0001_initial
 [X] 0002_auto_20180124_2311
 [X] 0003_auto_20181005_1415
 [X] 0004_auto_20181005_1415
 [X] 0005_auto_20181005_1445
modoboa_postfix_autoreply
 [X] 0001_initial
 [X] 0002_auto_20150728_1236
 [X] 0003_move_aliases
 [X] 0004_delete_alias
 [X] 0005_auto_20151202_1623
 [X] 0006_auto_20160329_1501
 [X] 0007_auto_20180928_1423
 [X] 0008_delete_transport
modoboa_radicale
 [X] 0001_initial
 [X] 0002_auto_20170831_1721
 [X] 0003_auto_20180324_1834
relaydomains
 [X] 0001_initial
 [X] 0002_migrate_from_modoboa_admin_relaydomains
 [X] 0003_auto_20170116_1508
 [X] 0004_auto_20161105_1424
 [X] 0005_auto_20161105_1426
 [X] 0006_auto_20170215_0948
 [X] 0007_recipientaccess
 [X] 0008_auto_20171123_1653
 [X] 0009_auto_20171124_1508
reversion
 [X] 0001_squashed_0004_auto_20160611_1202 (4 squashed migrations)
sessions
 [X] 0001_initial
sites
 [X] 0001_initial
 [X] 0002_alter_domain_unique
transport
 [X] 0001_initial
 [X] 0002_auto_20180928_1520
```

**Comment by tonioo** [comment]
@PatTheMav The command you're using will not migrate data from your current postgres DB to a new mysql one, it will just create a new mysql database. There is no tool to automatically migrate a postgres database to a mysql one. Anyway, I don't understand why this specific migration script fails... Have you tried a fresh install using mysql instead of postgres?

**Comment by PatTheMav** [comment]
@tonioo In a way, I did:

> after manually copying over all tables and recreating accounts in permissions in MySQL, **Modoboa runs without issues on the new database as well**.

I used the `migrate` command to actually create the necessary tables so I could then use a `dumpdata` and `loaddata` cycle to import the actual data.

What I ended up doing was commenting out the postgres `default` database, replaced it with the `mysql` one and executed `migrate` again. That ran into another issue: Namely the later reverted change of a too-long `varchar`field in `modoboa_postfix_autoreply.0002_auto_20150728_1236`. I had to manually edit it inside `site-packages` to reflect the value it was later changed too.

After I fixed that issue, migration finished without further issues and I had a full Modoboa schema in my MySQL database. When loading a previously exported json dump, more issues occurred:

```
  File "/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa/limits/handlers.py", line 49, in create_user_limits
    global_params = dict(param_tools.get_global_parameters("limits"))
  File "/srv/modoboa/env/local/lib/python2.7/site-packages/modoboa/parameters/tools.py", line 202, in get_global_parameters
    return get_localconfig().parameters.get_values(app, **kwargs)
AttributeError: Problem installing fixture '/srv/modoboa/modoboa.json': 'NoneType' object has no attribute 'parameters'
```

Loading a json dump without `contenttypes` lead to relation issues (`ValueError: Problem installing fixture '/srv/modoboa/modoboa.json': Cannot assign "<ContentType: domain>": the current database router prevents this relation` and loading a complete json dump to even more obscure errors (`django.db.utils.IntegrityError: <exception str() failed>`).

---

I guess the bottom line is: Don't do this, if you don't know exactly what you're doing. I don't expect this to be working use case, this issue is more of an "FYI" to you guys.
